### PR TITLE
Query now expose indexer meta

### LINKED ISSUE #325: Expose indexer metadata though query service

### PR DESCRIPTION
**Pull request by jiqiang90**
Closes #325 

- Allow query service to query indexer metadata though add `--meta <meta-api>`

### CODE REVIEW

File: packages/query/src/yargs.ts [hunk 1]
```diff
@@ -33,6 +33,11 @@ export function getYargsOption() {
       default: 'info',
       choices: ['fatal', 'error', 'warn', 'info', 'debug', 'trace', 'silent'],
     },
+    meta: {
```
Comment (ianhe8x) [inline packages/query/src/yargs.ts:36]: `--meta` seems not very clear to me.
```suggestion
    indexer: {
```
and pass the root endpoint of indexer. You might need /health of indexer as well.

File: packages/query/src/graphql/plugins/GetMetadataPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        blockTime: Int
+      }
+      extend type Query {
+        Metadata: Metadata
+      }
+    `,
+    resolvers: {
+      Query: {
+        Metadata: async () => {
+          return fetch(`${metaUrl}`).then((res) => res.json());
```
Comment (ianhe8x) [inline packages/query/src/graphql/plugins/GetMetadataPlugin.ts:33]: merge the result with indexer's status.

File: packages/query/src/graphql/plugins/GetMetadataPlugin.ts [hunk 3]
```diff
@@ -3,10 +3,27 @@
 
 import {makeExtendSchemaPlugin, gql} from 'graphile-utils';
 import fetch from 'node-fetch';
+import {setAsyncInterval} from '../../utils/asyncInterval';
 import {argv} from '../../yargs';
 
 const metaUrl = argv('meta') as string | undefined;
 
+type MetaData = {
```
Comment (ianhe8x) [inline packages/query/src/graphql/plugins/GetMetadataPlugin.ts:11]: I think we only keep the following: `targetHeight`, `lastProcessedHeight`, `lastProcessedTimestamp`, `chain`, `specName`, `genesisHash`, `indexerHealthy`, `indexerNodeVersion`, `queryNodeVersion`

File: packages/query/src/graphql/plugins/GetMetadataPlugin.ts [hunk 4]
```diff
@@ -1,48 +1,45 @@
 // Copyright 2020-2021 OnFinality Limited authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
+import * as url from 'url';
```
Comment (ianhe8x) [inline packages/query/src/graphql/plugins/GetMetadataPlugin.ts:4]: `import url from 'url';`

File: packages/query/src/graphql/plugins/GetMetadataPlugin.ts [hunk 5]
```diff
@@ -1,48 +1,45 @@
...
 
-let metaCache: MetaData;
+const metaCache = {
+  queryNodeVersion: version,
+} as MetaData;
 
 export const GetMetadataPlugin = makeExtendSchemaPlugin((build) => {
   return {
     typeDefs: gql`
       type Metadata {
```
Comment (ianhe8x) [inline packages/query/src/graphql/plugins/GetMetadataPlugin.ts:33]: use `_Metadata` to differentiate it from user specified entities

Comment (jiqiang90) [reply packages/query/src/graphql/plugins/GetMetadataPlugin.ts:33]: Fixed

File: packages/query/src/graphql/plugins/GetMetadataPlugin.ts [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      },
+    },
+  };
+});
+
+setAsyncInterval(async () => {
+  let health;
+  let meta;
+  try {
+    meta = await fetch(url.resolve(indexerUrl, `meta`));
```
Comment (ianhe8x) [inline packages/query/src/graphql/plugins/GetMetadataPlugin.ts:60]: url.resolve is depricated

**Review by ianhe8x — APPROVED** [review]

**Pull request merged**